spring-cloud-contract-verifier/src/main/java/org/springframework/cloud/contract/verifier/messaging/kafka/ContractVerifierKafkaStubMessagesInitializer.java

```diff
@@ -29,27 +29,31 @@
 import org.springframework.kafka.test.EmbeddedKafkaBroker;
 import org.springframework.kafka.test.utils.KafkaTestUtils;
 
-class ContractVerifierKafkaStubMessagesInitializer
-		implements KafkaStubMessagesInitializer {
+class ContractVerifierKafkaStubMessagesInitializer implements KafkaStubMessagesInitializer {
 
-	private static final Log log = LogFactory
-			.getLog(ContractVerifierKafkaStubMessagesInitializer.class);
+	private static final Log log = LogFactory.getLog(ContractVerifierKafkaStubMessagesInitializer.class);
 
 	@Override
-	public Map<String, Consumer> initialize(EmbeddedKafkaBroker broker,
-			KafkaProperties kafkaProperties) {
+	public Map<String, Consumer> initialize(EmbeddedKafkaBroker broker, KafkaProperties kafkaProperties) {
 		Map<String, Consumer> map = new HashMap<>();
 		for (String topic : broker.getTopics()) {
 			map.put(topic, prepareListener(broker, topic, kafkaProperties));
 		}
 		return map;
 	}
 
-	private Consumer prepareListener(EmbeddedKafkaBroker broker, String destination,
-			KafkaProperties kafkaProperties) {
-		Map<String, Object> consumerProperties = KafkaTestUtils.consumerProps(
-				kafkaProperties.getConsumer().getGroupId(), "false", broker);
-		consumerProperties.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+	private Consumer prepareListener(EmbeddedKafkaBroker broker, String destination, KafkaProperties kafkaProperties) {
+		Map<String, Object> consumerProperties = KafkaTestUtils
+				.consumerProps(kafkaProperties.getConsumer().getGroupId(), "false", broker);
+
+		// Respect custom key/value deserializers and any additional props under
+		// 'spring.kafka.consumer.properties'
+		consumerProperties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
```

Comment:
Without this, `KafkaTestUtils.consumerProps` always configures in a `<Integer, String>` key/value serde.  I chose to not pull in all of the consumer configured `spring.kafka.consumer` properties though as over-configuration can lead to confusing behavior during the tests. This keeps it constrained like it was before, but opens the door just a tiny bit to allow them to set the serde for key/value.